pkg/endpoint/public_ip_watcher_internal_test.go

```diff
@@ -0,0 +1,182 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoint
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	fakeDynClient "github.com/submariner-io/admiral/pkg/fake"
+	"github.com/submariner-io/admiral/pkg/syncer/test"
+	"github.com/submariner-io/admiral/pkg/watcher"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	fakeClientset "github.com/submariner-io/submariner/pkg/client/clientset/versioned/fake"
+	submarinerClientsetv1 "github.com/submariner-io/submariner/pkg/client/clientset/versioned/typed/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/types"
+	"github.com/submariner-io/submariner/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+const (
+	UsingLoadBalancer = "using-loadbalancer"
+	initialIP         = "1.2.3.4"
+	updatedIP         = "4.3.2.1"
+	clusterID         = "eastcluster"
+)
+
+var _ = Describe("public ip watcher", func() {
+	stopCh := make(chan struct{})
+	var submSpec *types.SubmarinerSpecification
+	var watcherConfig *PublicIPWatcherConfig
+
+	t := newPublicIPWatcherTestDriver()
+
+	BeforeEach(func() {
+		submSpec = &types.SubmarinerSpecification{
```

Comment:
This is only used in this function so can be local.

```suggestion
		submSpec := &types.SubmarinerSpecification{
```